pkg/client/cli/cmd/list.go

```diff
@@ -119,8 +120,10 @@ func (s *listCommand) list(cmd *cobra.Command, _ []string) error {
 	ch := make(chan *connector.WorkloadInfoSnapshot)
 	go func() {
 		for {
-			r, err := stream.Recv()
+			var r *connector.WorkloadInfoSnapshot
+			r, err = stream.Recv()
```

Comment:
So `err` here was private because this is a separate go routine. Unless its private, you'll need to protect it with a mutex outside of this go routine (a context cancellation might lead to concurrent access). It would probably be cleaner to send both `r` and `err` in a struct on the channel.